src/query/storage/m3/storage.go

```diff
@@ -70,11 +72,17 @@ func NewStorage(
 		return nil, err
 	}
 
+	adminOpts := client.NewAdminOptions()
```

Comment:
This should probably take in existing admin options that have already had the transforms applied

Reply:
The existing `client.AdminOptions` don't get build from the opts until much further down in `NewAdminClient(...)` whereas here we need this further up in `NewStorage`. Is it worth building the `client.AdminOptions` once up at the top and pushing down to both of these places?

Reply:
Think it would be better to pass in the seriesiteratorprocessor more explicitly rather than through the admin options. It's too easy to get this wrong.